.github/workflows/build.yml

```diff
@@ -0,0 +1,65 @@
+# This workflow will run:
+# 1. install Python dependencies
+# 2. build python package
+# 3. run unit tests
+# 4. Build documentation
+
+
+name: Python Package
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build_and_test:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: ["3.10", "3.11", "3.12"]
```

Comment:
There is no python version `3.12` in `pyproject.toml` file. Do we want to support it? if so, it should be added to the `pyproject.toml `, otherwise let's remove it form here.

Reply:
I will added it to `pyproject.toml`